### DISCUSSION TITLE
Assign 10.xx.xx.xx IP address to outbound IGMP traffic and no PPPoE tag

### DISCUSSION THREAD

**Discussion by horolux** [fabianishere/udm-iptv#86] [Ideas]
Hi,

I don't know if it is a new feature or something already possible or maybe even complicated to implement, but I have the need that my outbound multicast traffic uses a 10.xx.xx.xx IP address as SRC address.  In order to get IPTV working for my provider the JOIN request has to be done with a SRC address 10.xx.xx.xx otherwise it won't work. Also this request must not have a PPPoE tag. 

I can explain you this in a more detailed way, but I cannot post detailed information here as it I'm not allowed to do so (got some information from the IPTV provider in order to get this working but any publishing of it is forbidden). 

So I have a PPPoE connection established by my UDM-SE. 
UDM-IPTV is running on it. 
The IGMP JOIN request is well routed from internal network to ppp0 but today the SRC address is my official IP Address and
it must be a specific 10.xx.xx.xx address in order to get the "handshake" done or let's say the JOIN request accepted. And as said
this Join request must to be PPPoE tagged.

**Comment by fabianishere** [comment]
10.0.0.0/8 belongs to the private IP address space, is your ISP able to route these packets back to you when sending with such a source?

Also, how do you determine what the 10.0.0.0/8 address is? Is it assigned to you?

Anyway, if you don’t want to replace the SRC field in the packet with your public address, I believe you need to look into bridged IPTV, as opposed to routed IPTV (which this project does).

**Reply by horolux** [reply]
I know that 10.0.0.0/8 belongs to the private IP address space ;-) It's very difficult to explain the exact functionality by text. Is there any possibility to setup a short Zoom meeting (I can host it) in order to show you the 5 packet screenshots which I am not allowed to publish? It would be clear after I think. 

In order to get a better understanding I've actually connected my AVM Fritzbox again (as on the Fritzbox it's working out of the box), connected my IPTV receiver to it and started taking traces, which is possible directly on the Fritzbox via a special page. First thing I saw is that even if I'm on a mono-vc DSL the Fritbox configuration contains 2 "interfaces":   internet and iptv.  I took a backup of the
Fritzbox configuration file (which I may provide you by private message), it's in clear text, and what I see so far is that mcupstream points to the defined "iptv" interface which has a 10.xx.xx.xx address configured as well as VLAN ID 35 (same as for the "internet" interface".

As said, it's very difficult, nearly impossible to explain it here by text, I would like to set up a zoom meeting and share my screen with you while explaining. Maybe it can be done with the correct UDM-IPTV setup (VLAN ID, iptv interface, ...etc )?

The 10.0.0.0/8 is a fixed IP I have to set. It's not assigned, it's a fix address that is used just for the multicast request. Before and after the multicast join, public IPs are used again.  

Idea:  if I configure today VLAN ID 35 to udm-iptv it creates an IPTV "device/container" but this one gets then no IP address. Would it be an idea to set the IP address for it manually to 10.10.10.10 (the one I need)? If yes or worth a try
how can I set it manually? 

Thanks.

**Reply by fabianishere** [reply]
I don't have the time to do Zoom sessions for this project. Please share the Fritzbox configuration, that might provide some information.

**Reply by reinierjh** [reply]
Wat you're asking is the default way this works.
The IPTV network gets a ipaddress (in my case also a 10.x.x.x) the masquarading (NAT) done by the router makes sure that the traffic is routed to the correct gateway/interface. If this is not working, than look into the routing. All traffic to the IGMP gateway should flow via the IPTV network/Interface. 
BTW it is never a good idea to use fixed/static ip adresses on wan/iptv connections. 

Maybe tell us wich internet provider you are useing? If you live in a EU country then the provider will give you the correct settings to implement a router of choice.
If all is impossible. tell us wich profile you used on the fritzbox. If it's one of the bakedin configs. This is easy to get from the router.

**Reply by horolux** [reply]
Hi,

Yes I live in an EU country. I have a very detailed support from my internet / IPTV provider that's why I know that that outbound multicast traffic needs to have a specific IP address 10.10.10.10 and no PPPoE tag.  As you write, I see that a lot of people have private network address ranges in their UDM-IPTV configurations that why I was opening this discussion. 

The difference is that I don't GET an IP address from my provider for my IPTV, it's more that in order to get the mulitcast "handshake" completed the JOIN must come from an (static) 10.10.10.10 address, and what I see is the request been done with my official IP address. So the IPTV receiver triggers the multicast JOIN on my internal network, IGMP Proxy captures it and I see the JOIN requested on my ppp0 device but with an official IP address.  I need it there with the 10.10.10.10 SRC address, without the PPPoE tag and VLAN ID 35.

If there is a possibility to send you some screenshots of packet traces (which show how it shall work) I can do so, it will be clear in a couple of minutes for you if you have deep network knowledge. Unfortunately I cannot post these screenshots on a public place. As this is a community, I can post the working result here if needed - but first I need some help to get it working ;-) I can also tell you more details concerning the Fritzbox configuration or maybe attach the file here (after cleaning it up).

**Reply by reinierjh** [reply]
It would be a lot easier if you could tell us which internet provider you use.
Being part of the EU means that you are entitled to use your own router. The internet provider must supply the data for manufacturers and end users to be able to set this up.

But if I'm guessing about your ISP given the data you provide, you could just use the UDM-IPTV package.
You haven't sent a screenshot of your config yet. I assume that internet is already working through the network application set up.

You can start with VLAN35, for WAN Ranges try 0.0.0.0/0
For DHCP options (found in the fritzbox's config) I'm guessing "-O staticroutes"
I then had to come up with a random MAC address (I chose eth9 (my highest interface) and then add 2 to the last part of the macaddress)
Enter this mac address in the config file /etc/udm-iptv.conf
on a new line IPTV_WAN_VLAN_MAC="xx:xx:xx:xx:xx:xx"
Then restart application (systemctl restart udm-iptv)
Then check if the iptv interface is created. (udm-iptv-diag is the easiest)

**Reply by fabianishere** [reply]
@horolux Most helpful for us would be the raw Fritzbox configuration. Since it works with that device, I expect it is possible to get it working on the UDM.

**Reply by horolux** [reply]
Hi,

Internet is working fine. I've got what I need to setup my internet connection with the UDM-SE instead of the Fritzbox. 
My Internet is a mono-vc DSL which needs VLAN ID 35 to be set in order to work. That's all fine and working. 

Concerning UDM-IPTV: as soon as I set any VLAN ID (so also for 35) it does not work as it gets no IP address via DHCP for the 
iptv device. 

As requested, I've attached the raw Fritzbox configuration file (cleaned up, but the important config is present). You see that choosing the profile for my ISP as well as the DSL setup (mono-vc having the given IPTV service on it), the Fritzbox configuration defines 2 interfaces (dslifaces), one called "internet" and one called "iptv" and you'll see that there is a keyword "mcupstream=iptv" which for me tells the router to use that interface for multicast upstream (=traffic going to the Internet).  And that's where 10.10.10.10. is configured. 

So, maybe I would have to set VLAN 35 for UDM-IPTV and then somehow set the IP address for iptv device used by UDM-IPTV manually to 10.10.10.10. ?  And my understanding is that today only DHCP is forseen in the project. I'm even not sure if that's all what I need as the IPTV provider told me also that the IGMP JOIN request done with 10.10.10.10 must not have a PPPoE tag.  That can be seen in the FB config looking at "dsl_encap" which is different for the "internet" and "iptv" interface defined.

[FRITZ.Box 7490 113.07.29i_05.03.22_0923_clean.export.txt](https://github.com/fabianishere/udm-iptv/files/8208738/FRITZ.Box.7490.113.07.29i_05.03.22_0923_clean.export.txt)
.

**Reply by fabianishere** [reply]
I recommend you test the following config:

```bash
IPTV_WAN_INTERFACE="eth8.35"
IPTV_WAN_VLAN="0"
IPTV_WAN_RANGES="0.0.0.0/0"
```

and then manually assign an IP address to the interface:

```bash
ip addr add 10.10.10.10/24 dev eth0
```

**Reply by horolux** [reply]
Hi,

I'll will try this, but I don't understand why eth8.35? Because it's the WAN1 Port of the UDM combined to VLAN ID 35? Why not ppp0?
If I add the IP do eth0 it will be added to port1 of the UDM-SE switch no (eth0-eth7 are the 8 ports of the internal switch?) ?

I just want to understand the idea behind it.

**Reply by horolux** [reply]
With eth8.35 it does not work at all. Sure it should be eth8.35 and eth0 in your proposal?

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8.35
WAN Ranges: 0.0.0.0/0
WAN VLAN: 0 (dev iptv)
WAN DHCP options: 
LAN Interfaces: br20
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Mar 09 19:29:37 UDM-SE systemd[1]: udm-iptv.service: Consumed 99ms CPU time.
Mar 09 19:29:37 UDM-SE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Mar 09 19:29:37 UDM-SE udm-iptv[10780]: NATing IPTV network ranges (if necessary)
Mar 09 19:29:37 UDM-SE udm-iptv[10780]: Setting up igmpproxy
Mar 09 19:29:37 UDM-SE udm-iptv[10780]: Starting igmpproxy
Mar 09 19:29:37 UDM-SE udm-iptv[10780]: adding VIF, Ix 0 Fl 0x0 IP 0x0114000a br20, Threshold: 1, Ratelimit: 0
Mar 09 19:29:37 UDM-SE udm-iptv[10780]: There must be at least 1 Vif as upstream.
Mar 09 19:29:37 UDM-SE systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Mar 09 19:29:37 UDM-SE systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Mar 09 19:29:37 UDM-SE systemd[1]: udm-iptv.service: Consumed 104ms CPU time.

**Reply by fabianishere** [reply]
You should add the IP address to `eth8.35`, not `eth0` (my mistake):

```bash
ip addr add 10.10.10.10/24 dev eth8.35
```

**Reply by fabianishere** [reply]
`eth8.35` and not `ppp0`, because your ISP says the packets shouldn’t be PPPoE tagged.

**Reply by horolux** [reply]
Hi, thanks for the help. In the meantime I tried with eth8.35 (add the IP to this instead of eth0) and at least UDM-IPTV seems to accept it and work:

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8.35
WAN Ranges: 0.0.0.0/0
WAN VLAN: 0 (dev iptv)
WAN DHCP options: 
LAN Interfaces: br20
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
28: eth8.35@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.10.10.10/32 scope global eth8.35
       valid_lft forever preferred_lft forever
=== Service Logs ===
Mar 09 19:35:02 UDM-SE udm-iptv[28238]: Joining group 224.0.0.251 on interface eth8.35
Mar 09 19:35:02 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.10.10.10     to 224.0.0.251
Mar 09 19:35:02 UDM-SE udm-iptv[28238]: The IGMP message was from myself. Ignoring.
Mar 09 19:35:03 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.0.10.22      to 224.0.0.251
Mar 09 19:35:03 UDM-SE udm-iptv[28238]: Mebership report was received on a disabled interface. Ignoring.
Mar 09 19:35:03 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.0.80.1       to 233.89.188.1
Mar 09 19:35:03 UDM-SE udm-iptv[28238]: The IGMP message was from myself. Ignoring.
Mar 09 19:35:05 UDM-SE udm-iptv[28238]: The IGMP message was local multicast. Ignoring.
Mar 09 19:35:06 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.10.10.10     to 224.0.0.251
Mar 09 19:35:06 UDM-SE udm-iptv[28238]: The IGMP message was from myself. Ignoring.

**Reply by horolux** [reply]
I'll now test it, I may have to add some FW rules too.

**Reply by horolux** [reply]
Very good news:  it seems to work !!  I'll do some more tests and if the full functionnality is given I'll share everything with you in order to add the configuration to the settings table on the mainpage of the project - this will add one more IPTV provider to the list. And of course I'll contribute to the project (buying a coffee ;-) )

By the way:  I see now the 172.19.9.4 which is the answering address I know from the information got from my IPTV provider.
What's the meaning of MFC ?

Now I'll have to see how to configure this in order to make it persistent (working after a reboot of the device), and I have some 
input for the Ubiquiti ticket ;-) 

=== Service Logs ===
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.0.20.154     to 224.11.2.5
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: Updated route entry for 224.11.2.5 on VIF#1
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: Adding MFC: 172.19.9.4 -> 224.11.2.5, InpVIf: 0
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.10.10.10     to 224.11.2.5
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: The IGMP message was from myself. Ignoring.
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.0.80.1       to 224.0.0.251
Mar 09 19:42:09 UDM-SE udm-iptv[28238]: The IGMP message was from myself. Ignoring.
Mar 09 19:42:10 UDM-SE udm-iptv[28238]: RECV V2 member report   from 10.0.1.241      to 224.0.0.251
Mar 09 19:42:10 UDM-SE udm-iptv[28238]: Mebership report was received on a disabled interface. Ignoring.
Mar 09 19:42:12 UDM-SE udm-iptv[28238]: The IGMP message was local multicast. Ignoring.

**Reply by fabianishere** [reply]
Awesome! 😀

MFC stands for Multicast forwarding cache and contains the multicast forwarding rules installed by `igmpproxy`. You could try to see if IPTV still works when setting `IPTV_WAN_RANGES` to `172.19.0.0/16`. With the current config, it will forward multicast traffic originating from any source. However, since the IPTV traffic is routed over a private network, the security risk is small.

**Reply by horolux** [reply]
Yep. One reason why the provider routes over a private network is security.  By the way, you've got some cups of coffee ;-) 
Let me know how (where) to share the wole configuration with you in order to add it to the table on the main page, I'll do that later or tomorrow.

**Reply by fabianishere** [reply]
Thanks! 

I suppose it is enough to add a new entry to the table in the README with the name of your provider, 0 for WAN VLAN, and 0.0.0.0/0 for WAN ranges. Then in the notes, I would say something like:

1. Set WAN interface to `eth8.35`
2. Assign IP address to this interface on boot: `ip addr add 10.10.10.10/24 dev eth8.35`

I will probably add some functionality to the script to perform the latter step automatically in the future.

**Reply by horolux** [reply]
I just deactivated my FW which is needed in order to get it working (as the traffic coming from 172.19.x.x is neither related nor established there needs to be a rule to let it pass), and the IP address on eth8.35 was gone. I had to set it again and restart everything.

Yes it would be very useful to add an option to UDM-IPTV for such a configuration as mine. 

So I need to provide you the exact IPTV provider name, VLAN ID, adress for WAN_Range (after some more tests), firewall rules needed etc. I'll send this in the next days after some more tests. How to you want the information? Here? Other place?

**Reply by fabianishere** [reply]
You can provide it here or create a pull request with the necessary information.

**Comment by horolux** [comment]
Hi, 

So after some tests and different FW setups, watching a couple of days let's move forward and get this added as another profile for this great project:

**1. General information** 

Provider: PostTV
Country: LU

**2. The needed information for the profile creation:** 

```
=== Configuration ===
WAN Interface: eth8.35
WAN Ranges: 172.19.9.4/32
WAN VLAN: 0 (dev iptv)
WAN DHCP options: 
LAN Interfaces: br20
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
28: eth8.35@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.10.10.10/32 scope global eth8.35
       valid_lft forever preferred_lft forever
=== Service Logs ===
```
=> this is for WAN1, I think if someone is using WAN2 it should be the same except eth9.35 instead of eth8.35

**3.  Additionnal configuration needed**

Maybe you can in the meantime set this automatically via the new static IP function?

if WAN1 used:
ip addr add 10.10.10.10/32 dev eth8.35 

if WAN2 used:
ip addr add 10.10.10.10/32 dev eth9.35

**4. Firewall rule** 

There is a need for a firewall rule as the incoming multicast stream is not covered by the default "established/related" firewall rule. So, as we need to add a rule, let's then try to not open more then
absolutely needed. 

(a) Create the following profiles:

- IPTV Multicast source        : choose IPv4 Address/Subnet and add 172.19.9.4 as address
- IPTV Multicast destination : choose IPv4 Address/Subnet and add 224.11.0.0/16 as address/subnet
- IPTV Ports                           :  choose Port Group and add 4098, 8008, 50938, 60036  as ports

(b) Create the following firwall rule on INTERNET IN (legacy GUI: WAN IN):

- Allow IPTV  :    
        choose "Accept"
        choose "UDP" for IPv4 Protocol
        
        Settings for source:
        choose Source Type "Port/IP Group"
        choose "IPTV Multicast source" for IPv4 Address Group
        choose "IPTV Ports" for "Port Group"
        
        Settings for destination:
        choose Source Type "Port/IP Group"
        choose "IPTV Multicast destination" for IPv4 Address Group
        choose "IPTV Ports" for "Port Group"

**5. Final remarks /questions**

If something's missing or not clear enough in order to get this added to the projet, let me know ;-)
Once added/implemented, just let me know, I'll then test the next release if it works ;-)

Actual issue:  every time I change a something via the GUI related to the firewall settings etc, the IP set for
eth8.35 is gone and I have to redo the procedure manually (add it back, restart UDM-IPTV) in order to get
it working again. Would be great if we get this fixed too as well as ensuring the functionnality after a reboot.

**Reply by horolux** [reply]
Hi Fabian, how shall we proceed to get this added?

**Reply by horolux** [reply]
@fabianishere hi, how can I create a pull request or how can we work together in order to integrade the above configuration as a profile into udm-iptv? As it needs to add a static IP address to a specific device (here eth8.35), it would also be great that (a) udm-iptv sets this address (defined via the profile) and checks regulary if it is still set, if not should set it. This in unfortunately necessary as each firewall or network change via the UDM GUI unconfigures the static IP address from eth8.35 so IPTV is not working and one has to reconfigure it manually.

**Reply by fabianishere** [reply]
@horolux I have created a pull request with the new profile: https://github.com/fabianishere/udm-iptv/pull/107

The project now also supports watching the network configuration for changes, so it should re-apply the IP address if it is removed.

**Reply by yreiser** [reply]
Tried this with my UDM-Pro this weekend. Works fine with my Post LU IPTV. Many thanks!!!
I didn't have to setup any firewall rules though, immediately worked without!?

**Comment by Redbullux** [comment]
Hello everyone,
Until recently, the Post LU (Luxembourg) configuration worked perfectly without even adding rules on the firewall. A few days ago, the provider Post Luxembourg updated its TV boxes and since then I no longer have any TV stream reception. Here is the result of the diagnostic command:

root@Dream-Machine-Special-Edition:~# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8.35
WAN VLAN: 0 (dev iptv mac d0:21:f9:XX:XX:XX)
WAN DHCP: false (options "-O staticroutes")
WAN Static IP: 10.10.10.10/32
WAN Ranges: 172.19.9.0/24
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
24: eth8.35@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state                                                           UP group default qlen 1000
    inet 10.10.10.10/32 scope global eth8.35
       valid_lft forever preferred_lft forever
=== Service Logs ===
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@208]: if_mtu = 1500
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@214]: if_falgs = 4163
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@178]: eth8.35's ipv4 address is 10.10.10.10
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@195]: if_index = 24
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@208]: if_mtu = 1500
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: DEBUG[init_int                                                          erface@214]: if_falgs = 4163
Jun 25 21:35:30 Dream-Machine-Special-Edition udm-iptvd[1166241]: ERROR[main@619                                                          ]: exiting.........
Jun 25 21:35:30 Dream-Machine-Special-Edition IMPROXY[1166241]: ERROR[main@619]:                                                           exiting.........
Jun 25 21:35:30 Dream-Machine-Special-Edition systemd[1]: udm-iptv.service: Main                                                           process exited, code=exited, status=1/FAILURE
Jun 25 21:35:30 Dream-Machine-Special-Edition systemd[1]: udm-iptv.service: Fail                                                          ed with result 'exit-code'.


Can you help me?

**Reply by bassopt** [reply]
Tenta com o quick leave activo no script !
